File: 0784-insert-into-a-binary-search-tree/0784-insert-into-a-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void insert(TreeNode* root , int val){
        if( val <= root->val){
            if (!root->left){
                TreeNode * newnode = new TreeNode(val);
                root->left = newnode;
                return ;
            }
            else{
                insert(root->left, val);
            }
        }
        else if(val > root->val){
            if(!root->right){
                TreeNode * newnode = new TreeNode(val);
                root->right = newnode;    
                return         ;  
            }
            else{
            insert(root->right,val);
            }
        }
    }
    TreeNode* insertIntoBST(TreeNode* root, int val) {
        if (!root){
            TreeNode * newnode = new TreeNode(val);
            root = newnode;
            return root;
        }
        insert(root, val);
        return root;
    }
};

File: 0784-insert-into-a-binary-search-tree/0784-insert-into-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        def insert(root,val):
            if val <= root.val:
                if not root.left:
                    new = TreeNode(val)
                    root.left = new
                    return 
                else:
                    insert(root.left,val)

            elif val > root.val:
                if not root.right:
                    new = TreeNode (val)
                    root.right = new
                else:
                    insert(root.right,val)
        if not root:
            newNode = TreeNode(val)
            root = newNode
            return root
        dum = root
        insert(root,val)
        return dum